File: app/src/main/java/com/veganbeauty/app/features/account/reward/AccountRewardDetailFragment.java
package com.veganbeauty.app.features.account.reward;

import android.graphics.Color;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.core.content.ContextCompat;

import com.google.android.material.dialog.MaterialAlertDialogBuilder;
import com.veganbeauty.app.R;
import com.veganbeauty.app.core.base.RootieFragment;
import com.veganbeauty.app.data.local.LocalJsonReader;
import com.veganbeauty.app.data.local.RootieDatabase;
import com.veganbeauty.app.data.local.dao.RewardPointDao;
import com.veganbeauty.app.data.repository.OrderRepository;
import com.veganbeauty.app.databinding.AccountRewardDetailFragmentBinding;

import java.util.List;

import kotlinx.coroutines.flow.FlowCollector;

public class AccountRewardDetailFragment extends RootieFragment {

    private static final String ARG_GIFT_ID = "arg_gift_id";
    private static final String ARG_TITLE = "arg_title";
    private static final String ARG_DESCRIPTION = "arg_description";
    private static final String ARG_COST = "arg_cost";
    private static final String ARG_EXPIRY = "arg_expiry";
    private static final String ARG_CODE = "arg_code";
    private static final String ARG_TYPE = "arg_type";
    private static final String ARG_IS_OWNED = "arg_is_owned";
    private static final String ARG_DB_ID = "arg_db_id";
    private static final String ARG_RANK_REQUIRED = "arg_rank_required";
    private static final String ARG_MIN_ORDER_VALUE = "arg_min_order_value";
    private static final String ARG_APPLICABLE_PRODUCTS = "arg_applicable_products";
    private static final String ARG_OFFER_TYPE = "arg_offer_type";
    private static final String ARG_PRODUCT_ID = "arg_product_id";
    private static final String ARG_DISCOUNT_VALUE = "arg_discount_value";

    public static int selectRewardTabOnResume = 0;

    private AccountRewardDetailFragmentBinding _binding;
    private OrderRepository repository;

    private String giftId = "";
    private String giftTitle = "";
    private String giftDescription = "";
    private int giftCost = 0;
    private String giftExpiryDate = "";
    private String giftCode = "";
    private String giftType = "";
    private boolean isOwned = false;
    private int dbId = -1;
    private String rankRequired = "Đồng";
    private int minOrderValue = 0;
    private String applicableProducts = "Tất cả sản phẩm";
    private String offerType = "fixed_amount";
    private String productId = null;
    private int discountValue = 0;

    private int currentPoints = 8500;

    public static AccountRewardDetailFragment newInstance(
            String giftId, String title, String description, int cost,
            String expiryDate, String code, String giftType, boolean isOwned,
            int dbId, String rankRequired, int minOrderValue, String applicableProducts,
            String offerType, String productId, int discountValue) {

        AccountRewardDetailFragment fragment = new AccountRewardDetailFragment();
        Bundle args = new Bundle();
        args.putString(ARG_GIFT_ID, giftId);
        args.putString(ARG_TITLE, title);
        args.putString(ARG_DESCRIPTION, description);
        args.putInt(ARG_COST, cost);
        args.putString(ARG_EXPIRY, expiryDate);
        args.putString(ARG_CODE, code);
        args.putString(ARG_TYPE, giftType);
        args.putBoolean(ARG_IS_OWNED, isOwned);
        args.putInt(ARG_DB_ID, dbId);
        args.putString(ARG_RANK_REQUIRED, rankRequired);
        args.putInt(ARG_MIN_ORDER_VALUE, minOrderValue);
        args.putString(ARG_APPLICABLE_PRODUCTS, applicableProducts);
        args.putString(ARG_OFFER_TYPE, offerType);
        args.putString(ARG_PRODUCT_ID, productId);
        args.putInt(ARG_DISCOUNT_VALUE, discountValue);
        fragment.setArguments(args);
        return fragment;
    }

    private AccountRewardDetailFragmentBinding getBinding() {
        return _binding;
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        _binding = AccountRewardDetailFragmentBinding.inflate(inflater, container, false);
        
        if (getArguments() != null) {
            giftId = getArguments().getString(ARG_GIFT_ID, "");
            giftTitle = getArguments().getString(ARG_TITLE, "");
            giftDescription = getArguments().getString(ARG_DESCRIPTION, "");
            giftCost = getArguments().getInt(ARG_COST, 0);
            giftExpiryDate = getArguments().getString(ARG_EXPIRY, "");
            giftCode = getArguments().getString(ARG_CODE, "");
            giftType = getArguments().getString(ARG_TYPE, "");
            isOwned = getArguments().getBoolean(ARG_IS_OWNED, false);
            dbId = getArguments().getInt(ARG_DB_ID, -1);
            rankRequired = getArguments().getString(ARG_RANK_REQUIRED, "Đồng");
            minOrderValue = getArguments().getInt(ARG_MIN_ORDER_VALUE, 0);
            applicableProducts = getArguments().getString(ARG_APPLICABLE_PRODUCTS, "Tất cả sản phẩm");
            offerType = getArguments().getString(ARG_OFFER_TYPE, "fixed_amount");
            productId = getArguments().getString(ARG_PRODUCT_ID);
            discountValue = getArguments().getInt(ARG_DISCOUNT_VALUE, 0);
        }

        setupRepository();
        return getBinding().getRoot();
    }

    private void setupRepository() {
        RootieDatabase db = RootieDatabase.getDatabase(requireContext());
        repository = new OrderRepository(
                db.orderDao(),
                db.rewardPointDao(),
                db.userGiftDao(),
                new LocalJsonReader(requireContext())
        );
    }

    @Override
    public void setupUI(@NonNull View view) {
        getBinding().btnBack.setOnClickListener(v -> getParentFragmentManager().popBackStack());

        getBinding().btnShare.setOnClickListener(v ->
                Toast.makeText(requireContext(), "Đã copy link chia sẻ quà tặng!", Toast.LENGTH_SHORT).show()
        );

        getBinding().tvGiftTitle.setText(giftTitle);
        getBinding().tvGiftSubtitle.setText(giftDescription);
        getBinding().tvGiftCost.setText(String.format("%,d xu", giftCost).replace(',', '.'));
        
        String displayDate = giftExpiryDate.contains(" ") ? giftExpiryDate.split(" ")[0] : giftExpiryDate;
        getBinding().tvExpiryDate.setText(displayDate);
        
        if (isOwned) {
            getBinding().tvUsageCodeChip.setText(giftCode);
        } else {
            getBinding().tvUsageCodeChip.setText("******");
        }
        
        String days;
        if (displayDate.equals("Hôm nay") || displayDate.equals("2026-06-11")) {
            days = "Hôm nay";
        } else if (displayDate.equals("2026-12-15") || displayDate.equals("2026-12-31")) {
            days = "Còn 6 tháng";
        } else if (displayDate.equals("2027-01-15")) {
            days = "Còn 7 tháng";
        } else {
            days = "Còn hiệu lực";
        }
        getBinding().tvRemainingDays.setText(days);

        if (days.equals("Hôm nay")) {
            getBinding().tvRemainingDays.setTextColor(ContextCompat.getColor(requireContext(), R.color.status_cancelled_text));
            getBinding().tvStatus.setText("Gấp");
        } else if (days.equals("Hết hạn")) {
            getBinding().tvRemainingDays.setTextColor(ContextCompat.getColor(requireContext(), R.color.gray_dark));
            getBinding().tvRemainingDays.setText("Hết hạn");
            getBinding().tvStatus.setText("Hết hiệu lực");
        } else {
            getBinding().tvRemainingDays.setTextColor(Color.parseColor("#7A9161"));
            getBinding().tvStatus.setText("Có hiệu lực");
        }

        getBinding().tvUsageLimit.setText(giftType.startsWith("voucher") ? "0/1" : "Chỉ nhận 1 lần");

        getBinding().tvRuleApplicableProducts.setText("Áp dụng cho: " + applicableProducts);
        if (minOrderValue > 0) {
            getBinding().tvRuleMinOrderValue.setText("Hóa đơn tối thiểu " + String.format("%,d đ", minOrderValue).replace(',', '.'));
        } else {
            getBinding().tvRuleMinOrderValue.setText("Áp dụng cho mọi hóa đơn");
        }
        
        switch (offerType) {
            case "percentage":
                getBinding().tvRuleOfferType.setText("Ưu đãi: Giảm " + discountValue + "%");
                break;
            case "fixed_amount":
                getBinding().tvRuleOfferType.setText("Ưu đãi: Giảm " + String.format("%,d đ", discountValue).replace(',', '.'));
                break;
            case "product_gift":
                getBinding().tvRuleOfferType.setText("Ưu đãi: Tặng kèm sản phẩm");
                break;
            default:
                getBinding().tvRuleOfferType.setText("Ưu đãi đặc quyền từ Rootie");
                break;
        }

        if (isOwned) {
            getBinding().btnRedeemNow.setText("Đã sở hữu");
            getBinding().btnRedeemNow.setEnabled(false);
            getBinding().btnRedeemNow.setBackgroundTintList(ContextCompat.getColorStateList(requireContext(), R.color.gray_light));
            getBinding().btnRedeemNow.setTextColor(ContextCompat.getColor(requireContext(), R.color.gray_dark));
            
            getBinding().btnRemove.setOnClickListener(v -> showRemoveConfirmationDialog());
            getBinding().btnRemove.setVisibility(View.VISIBLE);
        } else {
            getBinding().btnRemove.setVisibility(View.GONE);

            getBinding().btnRedeemNow.setOnClickListener(v -> handleRedeemAction());
        }
    }

    @Override
    public void observeViewModel() {
        RootieDatabase db = RootieDatabase.getDatabase(requireContext());
        androidx.lifecycle.FlowLiveDataConversions.asLiveData(db.rewardPointDao().getTotalPointsFlow())
            .observe(getViewLifecycleOwner(), ptsList -> {
                int pointsVal = (ptsList != null && !ptsList.isEmpty()) ? ptsList.get(0).total : 0;
                currentPoints = pointsVal;
                if (_binding != null) {
                    _binding.tvUserBalance.setText("Bạn hiện có: " + String.format("%,d xu", pointsVal).replace(',', '.'));

                    boolean isLockedByRank = (rankRequired.equals("Vàng") && pointsVal < 10000) || 
                                             (rankRequired.equals("VIP") && pointsVal < 20000) ||
                                             (rankRequired.equals("Kim Cương") && pointsVal < 20000);
                    boolean isNotEnoughPoints = pointsVal < giftCost;

                    if (pointsVal >= giftCost) {
                        _binding.pbRedeemProgress.setProgress(100);
                        _binding.tvQualifyStatus.setText("✓ Đủ điều kiện");
                        _binding.tvQualifyStatus.setTextColor(ContextCompat.getColor(requireContext(), R.color.primary));
                    } else {
                        _binding.pbRedeemProgress.setProgress(pointsVal * 100 / giftCost);
                        _binding.tvQualifyStatus.setText("✗ Chưa đủ xu");
                        _binding.tvQualifyStatus.setTextColor(ContextCompat.getColor(requireContext(), R.color.status_cancelled_text));
                    }

                    if (!isOwned) {
                        if (isLockedByRank) {
                            _binding.btnRedeemNow.setEnabled(false);
                            _binding.btnRedeemNow.setText("Chưa đủ hạng");
                            _binding.btnRedeemNow.setBackgroundTintList(ContextCompat.getColorStateList(requireContext(), R.color.gray_light));
                            _binding.btnRedeemNow.setTextColor(ContextCompat.getColor(requireContext(), R.color.gray_dark));
                        } else if (isNotEnoughPoints) {
                            _binding.btnRedeemNow.setEnabled(false);
                            _binding.btnRedeemNow.setText("Không đủ xu");
                            _binding.btnRedeemNow.setBackgroundTintList(ContextCompat.getColorStateList(requireContext(), R.color.gray_light));
                            _binding.btnRedeemNow.setTextColor(ContextCompat.getColor(requireContext(), R.color.gray_dark));
                        } else {
                            _binding.btnRedeemNow.setEnabled(true);
                            _binding.btnRedeemNow.setText("Đổi quà");
                            _binding.btnRedeemNow.setBackgroundTintList(ContextCompat.getColorStateList(requireContext(), R.color.primary));
                            _binding.btnRedeemNow.setTextColor(Color.WHITE);
                        }
                    }
                }
            });
    }

    private void handleRedeemAction() {
        if (currentPoints < giftCost) {
            Toast.makeText(requireContext(), "Bạn không đủ xu để đổi quà tặng này!", Toast.LENGTH_SHORT).show();
            return;
        }

        new Thread(() -> {
            try {
                boolean success = repository.redeemGift(
                        giftId, giftTitle, giftDescription, giftCost, giftExpiryDate, giftCode,
                        giftType, minOrderValue, applicableProducts, offerType, productId, discountValue
                );

                if (getActivity() != null) {
                    getActivity().runOnUiThread(() -> {
                        if (success) {
                            selectRewardTabOnResume = 1;

                            new MaterialAlertDialogBuilder(requireContext())
                                    .setTitle("Đổi quà thành công!")
                                    .setMessage("Chúc mừng! Bạn đã đổi thành công " + giftTitle + ". Quà tặng đã được lưu vào mục 'Quà của tôi'.")
                                    .setPositiveButton("Xem danh sách quà", (dialog, which) -> getParentFragmentManager().popBackStack())
                                    .setCancelable(false)
                                    .show();
                        } else {
                            Toast.makeText(requireContext(), "Đổi quà thất bại. Vui lòng thử lại!", Toast.LENGTH_SHORT).show();
                        }
                    });
                }
            } catch (Exception e) {
                e.printStackTrace();
            }
        }).start();
    }

    private void showRemoveConfirmationDialog() {
        new MaterialAlertDialogBuilder(requireContext())
                .setTitle("Xoá quà tặng")
                .setMessage("Bạn có chắc chắn muốn xoá " + giftTitle + " khỏi danh sách quà của tôi không?")
                .setPositiveButton("Xác nhận", (dialog, which) -> {
                    new Thread(() -> {
                        try {
                            boolean deleted = repository.deleteUserGiftById(dbId);
                            if (getActivity() != null) {
                                getActivity().runOnUiThread(() -> {
                                    if (deleted) {
                                        Toast.makeText(requireContext(), "Đã xoá quà tặng khỏi danh sách!", Toast.LENGTH_SHORT).show();
                                        getParentFragmentManager().popBackStack();
                                    } else {
                                        Toast.makeText(requireContext(), "Xoá quà tặng thất bại!", Toast.LENGTH_SHORT).show();
                                    }
                                });
                            }
                        } catch (Exception e) {
                            e.printStackTrace();
                        }
                    }).start();
                })
                .setNegativeButton("Hủy", null)
                .show();
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        _binding = null;
    }
}

File: app/src/main/java/com/veganbeauty/app/features/account/reward/AccountRewardDetailFragment.kt
package com.veganbeauty.app.features.account.reward

import android.graphics.Color
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.core.content.ContextCompat
import androidx.lifecycle.lifecycleScope
import com.google.android.material.dialog.MaterialAlertDialogBuilder
import com.veganbeauty.app.R
import com.veganbeauty.app.core.base.RootieFragment
import com.veganbeauty.app.data.local.LocalJsonReader
import com.veganbeauty.app.data.local.RootieDatabase
import com.veganbeauty.app.data.repository.OrderRepository
import com.veganbeauty.app.databinding.AccountRewardDetailFragmentBinding
import kotlinx.coroutines.launch

class AccountRewardDetailFragment : RootieFragment() {

    private var _binding: AccountRewardDetailFragmentBinding? = null
    private val binding get() = _binding!!

    private lateinit var repository: OrderRepository

    private var giftId = ""
    private var giftTitle = ""
    private var giftDescription = ""
    private var giftCost = 0
    private var giftExpiryDate = ""
    private var giftCode = ""
    private var giftType = ""
    private var isOwned = false
    private var dbId = -1
    private var rankRequired = "Đồng"
    private var minOrderValue = 0
    private var applicableProducts = "Tất cả sản phẩm"
    private var offerType = "fixed_amount"
    private var productId: String? = null
    private var discountValue = 0

    private var currentPoints = 8500

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _binding = AccountRewardDetailFragmentBinding.inflate(inflater, container, false)
        
        arguments?.let {
            giftId = it.getString(ARG_GIFT_ID) ?: ""
            giftTitle = it.getString(ARG_TITLE) ?: ""
            giftDescription = it.getString(ARG_DESCRIPTION) ?: ""
            giftCost = it.getInt(ARG_COST)
            giftExpiryDate = it.getString(ARG_EXPIRY) ?: ""
            giftCode = it.getString(ARG_CODE) ?: ""
            giftType = it.getString(ARG_TYPE) ?: ""
            isOwned = it.getBoolean(ARG_IS_OWNED)
            dbId = it.getInt(ARG_DB_ID)
            rankRequired = it.getString(ARG_RANK_REQUIRED) ?: "Đồng"
            minOrderValue = it.getInt(ARG_MIN_ORDER_VALUE)
            applicableProducts = it.getString(ARG_APPLICABLE_PRODUCTS) ?: "Tất cả sản phẩm"
            offerType = it.getString(ARG_OFFER_TYPE) ?: "fixed_amount"
            productId = it.getString(ARG_PRODUCT_ID)
            discountValue = it.getInt(ARG_DISCOUNT_VALUE)
        }

        setupRepository()
        return binding.root
    }

    private fun setupRepository() {
        val db = RootieDatabase.getDatabase(requireContext())
        repository = OrderRepository(
            db.orderDao(),
            db.rewardPointDao(),
            db.userGiftDao(),
            LocalJsonReader(requireContext())
        )
    }

    override fun setupUI(view: View) {
        binding.btnBack.setOnClickListener {
            parentFragmentManager.popBackStack()
        }

        binding.btnShare.setOnClickListener {
            Toast.makeText(requireContext(), "Đã copy link chia sẻ quà tặng!", Toast.LENGTH_SHORT).show()
        }

        // Configure ticket view details
        binding.tvGiftTitle.text = giftTitle
        binding.tvGiftSubtitle.text = giftDescription
        binding.tvGiftCost.text = String.format("%,d xu", giftCost).replace(',', '.')
        
        // Expiry date format split
        val displayDate = if (giftExpiryDate.contains(" ")) giftExpiryDate.split(" ")[0] else giftExpiryDate
        binding.tvExpiryDate.text = displayDate
        
        // Bind the code chip (Mask code if not owned!)
        if (isOwned) {
            binding.tvUsageCodeChip.text = giftCode
        } else {
            binding.tvUsageCodeChip.text = "******"
        }
        
        val days = when (displayDate) {
            "Hôm nay", "2026-06-11" -> "Hôm nay"
            "2026-12-15" -> "Còn 6 tháng"
            "2026-12-31" -> "Còn 6 tháng"
            "2027-01-15" -> "Còn 7 tháng"
            else -> "Còn hiệu lực"
        }
        binding.tvRemainingDays.text = days

        if (days == "Hôm nay") {
            binding.tvRemainingDays.setTextColor(ContextCompat.getColor(requireContext(), R.color.status_cancelled_text))
            binding.tvStatus.text = "Gấp"
        } else if (days == "Hết hạn") {
            binding.tvRemainingDays.setTextColor(ContextCompat.getColor(requireContext(), R.color.gray_dark))
            binding.tvRemainingDays.text = "Hết hạn"
            binding.tvStatus.text = "Hết hiệu lực"
        } else {
            binding.tvRemainingDays.setTextColor(Color.parseColor("#7A9161"))
            binding.tvStatus.text = "Có hiệu lực"
        }

        binding.tvUsageLimit.text = if (giftType.startsWith("voucher")) "0/1" else "Chỉ nhận 1 lần"

        // Setup dynamic applicability checkmark rules
        binding.tvRuleApplicableProducts.text = "Áp dụng cho: $applicableProducts"
        binding.tvRuleMinOrderValue.text = if (minOrderValue > 0) {
            "Hóa đơn tối thiểu " + String.format("%,d đ", minOrderValue).replace(',', '.')
        } else {
            "Áp dụng cho mọi hóa đơn"
        }
        
        binding.tvRuleOfferType.text = when (offerType) {
            "percentage" -> "Ưu đãi: Giảm $discountValue%"
            "fixed_amount" -> "Ưu đãi: Giảm " + String.format("%,d đ", discountValue).replace(',', '.')
            "product_gift" -> "Ưu đãi: Tặng kèm sản phẩm"
            else -> "Ưu đãi đặc quyền từ Rootie"
        }

        // Setup owned details vs exchange details
        if (isOwned) {
            binding.btnRedeemNow.text = "Đã sở hữu"
            binding.btnRedeemNow.isEnabled = false
            binding.btnRedeemNow.backgroundTintList = ContextCompat.getColorStateList(requireContext(), R.color.gray_light)
            binding.btnRedeemNow.setTextColor(ContextCompat.getColor(requireContext(), R.color.gray_dark))
            
            val removeAction = View.OnClickListener {
                showRemoveConfirmationDialog()
            }
            binding.btnRemove.setOnClickListener(removeAction)
            binding.btnDelete.setOnClickListener(removeAction)
            
            binding.btnRemove.visibility = View.VISIBLE
            binding.btnDelete.visibility = View.VISIBLE
        } else {
            binding.btnRemove.visibility = View.GONE
            binding.btnDelete.visibility = View.GONE
            
            binding.btnRedeemNow.setOnClickListener {
                handleRedeemAction()
            }
        }
    }

    override fun observeViewModel() {
        val db = RootieDatabase.getDatabase(requireContext())
        viewLifecycleOwner.lifecycleScope.launch {
            db.rewardPointDao().getTotalPointsFlow().collect { points ->
                val pointsVal = points ?: 0
                currentPoints = pointsVal
                binding.tvUserBalance.text = "Bạn hiện có: ${String.format("%,d xu", pointsVal).replace(',', '.')}"
 
                val isLockedByRank = (rankRequired == "Vàng" && pointsVal < 10000) || 
                                     (rankRequired == "VIP" && pointsVal < 20000) ||
                                     (rankRequired == "Kim Cương" && pointsVal < 20000)
                val isNotEnoughPoints = pointsVal < giftCost
 
                if (pointsVal >= giftCost) {
                    binding.pbRedeemProgress.progress = 100
                    binding.tvQualifyStatus.text = "✓ Đủ điều kiện"
                    binding.tvQualifyStatus.setTextColor(ContextCompat.getColor(requireContext(), R.color.primary))
                } else {
                    binding.pbRedeemProgress.progress = (pointsVal * 100 / giftCost)
                    binding.tvQualifyStatus.text = "✗ Chưa đủ xu"
                    binding.tvQualifyStatus.setTextColor(ContextCompat.getColor(requireContext(), R.color.status_cancelled_text))
                }
 
                if (!isOwned) {
                    if (isLockedByRank) {
                        binding.btnRedeemNow.isEnabled = false
                        binding.btnRedeemNow.text = "Chưa đủ hạng"
                        binding.btnRedeemNow.backgroundTintList = ContextCompat.getColorStateList(requireContext(), R.color.gray_light)
                        binding.btnRedeemNow.setTextColor(ContextCompat.getColor(requireContext(), R.color.gray_dark))
                    } else if (isNotEnoughPoints) {
                        binding.btnRedeemNow.isEnabled = false
                        binding.btnRedeemNow.text = "Không đủ xu"
                        binding.btnRedeemNow.backgroundTintList = ContextCompat.getColorStateList(requireContext(), R.color.gray_light)
                        binding.btnRedeemNow.setTextColor(ContextCompat.getColor(requireContext(), R.color.gray_dark))
                    } else {
                        binding.btnRedeemNow.isEnabled = true
                        binding.btnRedeemNow.text = "Đổi quà"
                        binding.btnRedeemNow.backgroundTintList = ContextCompat.getColorStateList(requireContext(), R.color.primary)
                        binding.btnRedeemNow.setTextColor(Color.WHITE)
                    }
                }
            }
        }
    }

    private fun handleRedeemAction() {
        if (currentPoints < giftCost) {
            Toast.makeText(requireContext(), "Bạn không đủ xu để đổi quà tặng này!", Toast.LENGTH_SHORT).show()
            return
        }

        lifecycleScope.launch {
            val success = repository.redeemGift(
                giftId = giftId,
                title = giftTitle,
                description = giftDescription,
                cost = giftCost,
                expiryDate = giftExpiryDate,
                code = giftCode,
                giftType = giftType,
                minOrderValue = minOrderValue,
                applicableProducts = applicableProducts,
                offerType = offerType,
                productId = productId,
                discountValue = discountValue
            )

            if (success) {
                selectRewardTabOnResume = 1 // Redirect to My Gifts tab

                MaterialAlertDialogBuilder(requireContext())
                    .setTitle("Đổi quà thành công!")
                    .setMessage("Chúc mừng! Bạn đã đổi thành công $giftTitle. Quà tặng đã được lưu vào mục 'Quà của tôi'.")
                    .setPositiveButton("Xem danh sách quà") { _, _ ->
                        parentFragmentManager.popBackStack()
                    }
                    .setCancelable(false)
                    .show()
            } else {
                Toast.makeText(requireContext(), "Đổi quà thất bại. Vui lòng thử lại!", Toast.LENGTH_SHORT).show()
            }
        }
    }

    private fun showRemoveConfirmationDialog() {
        MaterialAlertDialogBuilder(requireContext())
            .setTitle("Xoá quà tặng")
            .setMessage("Bạn có chắc chắn muốn xoá $giftTitle khỏi danh sách quà của tôi không?")
            .setPositiveButton("Xác nhận") { _, _ ->
                lifecycleScope.launch {
                    val deleted = repository.deleteUserGiftById(dbId)
                    if (deleted) {
                        Toast.makeText(requireContext(), "Đã xoá quà tặng khỏi danh sách!", Toast.LENGTH_SHORT).show()
                        parentFragmentManager.popBackStack()
                    } else {
                        Toast.makeText(requireContext(), "Xoá quà tặng thất bại!", Toast.LENGTH_SHORT).show()
                    }
                }
            }
            .setNegativeButton("Hủy", null)
            .show()
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }

    companion object {
        private const val ARG_GIFT_ID = "arg_gift_id"
        private const val ARG_TITLE = "arg_title"
        private const val ARG_DESCRIPTION = "arg_description"
        private const val ARG_COST = "arg_cost"
        private const val ARG_EXPIRY = "arg_expiry"
        private const val ARG_CODE = "arg_code"
        private const val ARG_TYPE = "arg_type"
        private const val ARG_IS_OWNED = "arg_is_owned"
        private const val ARG_DB_ID = "arg_db_id"
        private const val ARG_RANK_REQUIRED = "arg_rank_required"
        private const val ARG_MIN_ORDER_VALUE = "arg_min_order_value"
        private const val ARG_APPLICABLE_PRODUCTS = "arg_applicable_products"
        private const val ARG_OFFER_TYPE = "arg_offer_type"
        private const val ARG_PRODUCT_ID = "arg_product_id"
        private const val ARG_DISCOUNT_VALUE = "arg_discount_value"

        var selectRewardTabOnResume = 0

        fun newInstance(
            giftId: String,
            title: String,
            description: String,
            cost: Int,
            expiryDate: String,
            code: String,
            giftType: String,
            isOwned: Boolean,
            dbId: Int,
            rankRequired: String,
            minOrderValue: Int,
            applicableProducts: String,
            offerType: String,
            productId: String?,
            discountValue: Int
        ): AccountRewardDetailFragment {
            val fragment = AccountRewardDetailFragment()
            val args = Bundle()
            args.putString(ARG_GIFT_ID, giftId)
            args.putString(ARG_TITLE, title)
            args.putString(ARG_DESCRIPTION, description)
            args.putInt(ARG_COST, cost)
            args.putString(ARG_EXPIRY, expiryDate)
            args.putString(ARG_CODE, code)
            args.putString(ARG_TYPE, giftType)
            args.putBoolean(ARG_IS_OWNED, isOwned)
            args.putInt(ARG_DB_ID, dbId)
            args.putString(ARG_RANK_REQUIRED, rankRequired)
            args.putInt(ARG_MIN_ORDER_VALUE, minOrderValue)
            args.putString(ARG_APPLICABLE_PRODUCTS, applicableProducts)
            args.putString(ARG_OFFER_TYPE, offerType)
            args.putString(ARG_PRODUCT_ID, productId)
            args.putInt(ARG_DISCOUNT_VALUE, discountValue)
            fragment.arguments = args
            return fragment
        }
    }
}
